File: backend/src/main/scala/microservices/order/objects/OrderPlannerResponse.scala
package com.typesafe.travel.order.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class OrderPlannerResponse(
    orderId: String,
    buyerUserId: String,
    orderType: String,
    status: String,
    orderCurrency: String,
    totalPrice: String,
    totalCapturedAmount: String,
    totalSettledRefundAmount: String,
    remainingRefundableAmount: String,
    createdAt: String,
    paidAt: Option[String],
    confirmedAt: Option[String],
    completedAt: Option[String],
    cancelledAt: Option[String],
    orderLineItems: List[OrderLineItemPlannerResponse],
    orderPayments: List[PaymentPlannerResponse],
    orderRefunds: List[RefundPlannerResponse]
)
object OrderPlannerResponse:
  given Encoder[OrderPlannerResponse] = deriveEncoder
  given Decoder[OrderPlannerResponse] = deriveDecoder

File: frontend/src/microservices/order/objects/OrderPlannerResponse.ts
// 本文件定义订单详情响应，和后端 `OrderPlannerResponse` 保持同名镜像。

import type { OrderLineItemPlannerResponse } from './OrderLineItemPlannerResponse'
import type { PaymentPlannerResponse } from './PaymentPlannerResponse'
import type { RefundPlannerResponse } from './RefundPlannerResponse'

export type OrderPlannerResponse = {
  orderId: string
  buyerUserId: string
  orderType: string
  status: string
  orderCurrency: string
  totalPrice: string
  totalCapturedAmount: string
  totalSettledRefundAmount: string
  remainingRefundableAmount: string
  createdAt: string
  paidAt: string | null
  confirmedAt: string | null
  completedAt: string | null
  cancelledAt: string | null
  orderLineItems: OrderLineItemPlannerResponse[]
  orderPayments: PaymentPlannerResponse[]
  orderRefunds: RefundPlannerResponse[]
}

export const orderPlannerResponseFromJson = (json: string): OrderPlannerResponse =>
  JSON.parse(json) as OrderPlannerResponse

export const orderPlannerResponseToJson = (value: OrderPlannerResponse): string =>
  JSON.stringify(value)
